### PR TITLE
Log dispatched actions

### PR DESCRIPTION
**Pull request by glennsl**
This adds `[@deriving show]` to `Actions.t` and dependent types, and sets up logging of the actions in the `dispatch` function in `StoreThread`.

There's quite a bit of `[@opaque]` here, but I expect those to mostly go away and for the actions to become more informative as we move away from actions as functions.

### CODE REVIEW

**Comment by glennsl** [comment]
Come on, Windows, u can do dis!

File: src/Model/Actions.re [hunk 1]
```diff
@@ -10,14 +10,15 @@ open Oni_Input;
 open Oni_Extensions;
 open Oni_Syntax;
 
+[@deriving show({with_path: false})]
 type t =
   | Init
   | Tick(tick)
   | BufferDisableSyntaxHighlighting(int)
-  | BufferEnter(Vim.BufferMetadata.t, option(string))
+  | BufferEnter([@opaque] Vim.BufferMetadata.t, option(string))
```
Comment (bryphe) [inline src/Model/Actions.re]: Nice find on `@opaque` - I didn't know this was available!

File: src/Store/StoreThread.re [hunk 2]
```diff
@@ -200,6 +200,12 @@ let start =
     SearchStoreConnector.subscriptions(ripgrep);
 
   let rec dispatch = (action: Model.Actions.t) => {
+    switch (action) {
+    | Tick(_) => () // This gets a bit intense, so ignore it
+    | _ =>
+      Core.Log.info("[SotreThread.dispatch]: " ++ Model.Actions.show(action))
```
Comment (bryphe) [inline src/Store/StoreThread.re]: Minor typo: `Sotre` -> `Store`

Great to have this hooked up!

File: src/Store/StoreThread.re [hunk 3]
```diff
@@ -200,6 +200,12 @@ let start =
     SearchStoreConnector.subscriptions(ripgrep);
 
   let rec dispatch = (action: Model.Actions.t) => {
+    switch (action) {
+    | Tick(_) => () // This gets a bit intense, so ignore it
```
Comment (bryphe) [inline src/Store/StoreThread.re]: Hopefully going away soon...

**Pull request merged**